src/main/scala/com/wavesplatform/Application.scala

```diff
@@ -39,80 +38,92 @@ import scorex.waves.http.{DebugApiRoute, WavesApiRoute}
-    wallet, stateReader, settings.blockchainSettings, utxStorage, coordinator)), "BlockGenerator")
-  lazy val blockchainSynchronizer: ActorRef = actorSystem.actorOf(Props(new BlockchainSynchronizer(networkController, coordinator, history, settings.synchronizationSettings)), "BlockchainSynchronizer")
-  lazy val peerSynchronizer: ActorRef = actorSystem.actorOf(Props(new PeerSynchronizer(networkController, peerManager, settings.networkSettings)), "PeerSynchronizer")
+  private lazy val upnp = new UPnP(settings.networkSettings.uPnPSettings) // don't initialize unless enabled
+  private val wallet: Wallet = Wallet(settings.walletSettings)
 
   def run(): Unit = {
     log.debug(s"Available processors: ${Runtime.getRuntime.availableProcessors}")
     log.debug(s"Max memory available: ${Runtime.getRuntime.maxMemory}")
 
     checkGenesis()
 
-    if (settings.networkSettings.uPnPSettings.enable) upnp.addPort(settings.networkSettings.port)
+    val feeCalculator = new FeeCalculator(settings.feesSettings)
+    val time: Time = new TimeImpl()
+    val utxStorage: UnconfirmedTransactionsStorage = new UnconfirmedTransactionsDatabaseImpl(settings.utxSettings.size)
+    val newTransactionHandler = new NewTransactionHandlerImpl(settings.blockchainSettings.functionalitySettings,
+      time, feeCalculator, utxStorage, stateReader)
+
+    val peerDatabase = new PeerDatabaseImpl(settings.networkSettings)
+    val establishedConnections = new ConcurrentHashMap[Channel, PeerInfo]
+    val allChannels = new DefaultChannelGroup(GlobalEventExecutor.INSTANCE)
+
+    val network = new NetworkServer(
+      settings.blockchainSettings.addressSchemeCharacter,
+      settings.networkSettings.bindAddress,
+      settings.networkSettings.declaredAddress,
+      settings,
+      history,
+      checkpoints,
+      blockchainUpdater,
+      time,
+      stateReader,
+      utxStorage,
+      newTransactionHandler,
+      peerDatabase,
+      wallet,
+      allChannels,
+      establishedConnections)
+
+    val apiRoutes = Seq(
+      BlocksApiRoute(settings.restAPISettings, settings.checkpointsSettings, history, checkpoint => network.writeToLocalChannel(checkpoint)),
+      TransactionsApiRoute(settings.restAPISettings, stateReader, history, utxStorage),
+      NxtConsensusApiRoute(settings.restAPISettings, stateReader, history, settings.blockchainSettings.functionalitySettings),
+      WalletApiRoute(settings.restAPISettings, wallet),
+      PaymentApiRoute(settings.restAPISettings, wallet, newTransactionHandler, time),
+      UtilsApiRoute(settings.restAPISettings),
+      PeersApiRoute(settings.restAPISettings, network.connect, peerDatabase, establishedConnections),
+      AddressApiRoute(settings.restAPISettings, wallet, stateReader, settings.blockchainSettings.functionalitySettings),
+      DebugApiRoute(settings.restAPISettings, wallet, stateReader, history, peerDatabase, blockId => network.writeToLocalChannel(RollbackTo(blockId))),
```

Comment:
Using lambda-senders instead of real dependencies surely make code less verbose, but also less explicit. 
It is disturbing that you write to local channel instead of just invoking rollbackTo on blockchainUpdater. If it's necessary, it might indicate a poor abstraction, let's discuss this